core/chains/evm/logpoller/log_poller.go

```diff
@@ -134,7 +134,8 @@ type logPoller struct {
 	// Usually the only way to recover is to manually remove the offending logs and block from the database.
 	// LogPoller keeps running in infinite loop, so whenever the invalid state is removed from the database it should
 	// recover automatically without needing to restart the LogPoller.
-	finalityViolated *atomic.Bool
+	finalityViolated           *atomic.Bool
+	countBasedLogPruningActive *atomic.Bool
```

Comment:
nit/ These could be values, then the zero value will work without requiring initialization in the constructor.
```suggestion
	finalityViolated           atomic.Bool
	countBasedLogPruningActive atomic.Bool
```